app/models/event_role.rb

```diff
@@ -122,16 +122,13 @@ def assigned_to(options = {})
     end
   end
   
-  def sort_index 
-    Roles_All.each_index { |role_index| return role_index if Roles_All[role_index] == role }
-    return -1 
+  def sort_index
+    Roles_All.index self.role
   end
 
   # define the natural sorting order
-  def <=> (role)
-    return 1 if sort_index < 0
-    return -1 if role.sort_index < 0
-    return sort_index <=> role.sort_index   
+  def <=> (other_role)
+    sort_index <=> other_role.sort_index
```

Comment:
In the original, if `each_index` does not cause a `return` (role not found in role list), `sort_index` returns `-1`. So the `< 0` checks here are a safeguard from an unknown or unlisted role. Of course we don't want to have any such roles, but I figure this safeguard was put here for a reason. It looks like `index` returns `nil` if not found so we can swap these for `nil?` checks.